server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/WatchRequestConverter.java

```diff
@@ -81,6 +82,25 @@ public WatchRequest convertRequest(
         return new WatchRequest(lastKnownRevision, timeoutMillis, notifyEntryNotFound);
     }
 
+    @VisibleForTesting
+    String extractRevision(String ifNoneMatch) {
+        // Three below cases are valid. See https://github.com/line/centraldogma/issues/415
+        // - <revision> (for backward compatibility)
+        // - "<revision>"
+        // - W/"<revision>"
+        if (ifNoneMatch.length() > 2 && ifNoneMatch.charAt(0) == '"' &&
+            ifNoneMatch.charAt(ifNoneMatch.length() - 1) == '"') {
+            return ifNoneMatch.substring(1, ifNoneMatch.length() - 1);
+        }
+
+        if (ifNoneMatch.length() > 4 && ifNoneMatch.startsWith("W/\"") &&
+            ifNoneMatch.charAt(ifNoneMatch.length() - 1) == '"') {
+            return ifNoneMatch.substring(3, ifNoneMatch.length() - 1);
```

Comment:
nit: Extract `ifNoneMatch.length()` as a local variable? It is used in multiple places.

Reply:
@ikhoon , thank you. I revised. 🙂